src/auth/.gcb/builds/api_key_test/main.tf

```diff
@@ -0,0 +1,78 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+variable "project" {}
+
+data "google_project" "project" {
+}
+
+# The API Keys service is used to create an API key
+resource "google_project_service" "apikeys" {
+  project = var.project
+  service = "apikeys.googleapis.com"
+
+  timeouts {
+    create = "30m"
+    update = "40m"
+  }
+
+  disable_dependent_services = true
+}
+
+# The language service is used to verify the API key
+resource "google_project_service" "language" {
+  project = var.project
+  service = "language.googleapis.com"
+
+  timeouts {
+    create = "30m"
+    update = "40m"
+  }
+
+  disable_dependent_services = true
+}
+
+resource "google_apikeys_key" "test-api-key" {
+  name         = "test-key"
+  display_name = "Test API Key"
+  project      = var.project
+
+  # Restrict the API Key to the one RPC we use to verify the credentials.
+  restrictions {
+    api_targets {
+      service = "language.googleapis.com"
```

Comment:
If the key ever gets leaked somehow, having access to language API is powerful I think because the attacker can use it for free to translate whatever they want.
It is less useful for an attacker if the key can be used for something like only storage API in the test project.

Reply:
Hmm. Any other resourceful APIs that support API key?